services/search.service.js

```diff
@@ -63,15 +64,84 @@ function executeQuery(model, queryArray, page, limit, sort, sort_by, shouldExpan
+    }
+    return query.limit(limit).skip(limit * page)
+}
+
+
+/**
+ * @function executeQuery
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @returns {Promise<[Array]>}
+ * @description Builds and executes a search query based on a subset of mongodb
+ */
+function executeQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
+    return query.exec('find');
+}
+
+
+/**
+ * @function executeStatusAction
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @param {string} update the JSON string containing the keys and values to update to
+ * @returns {Promise<[Array]>}
+ * @description Builds and executes a status update based on a subset of mongodb
+ */
+function executeStatusAction(model, queryArray, page, limit, sort, sortBy, update, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
+    var update_obj = JSON.parse(update);
+    return query.updateMany({ $set: update_obj }).exec();
+}
+
+
+/**
+ * @function executeEmailAction
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @param {"Accepted"|"Waitlisted"|"Reminder"} status the status type of the email to send to hackers
+ * @returns {Promise<[Array]>}
+ * @description Sends a status update email based on a subset of mongodb
+ */
+async function executeEmailAction(model, queryArray, page, limit, sort, sortBy, status, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
+    const hackers = await query.exec()
+
+    if (hackers) {
+        for (const hacker of hackers) {
+            const account = await AccountService.findById(hacker.accountId);
+            if (!account) {
+                break;
+            }
+            let emailError = await EmailService.sendStatusUpdateAsync(account.firstName, account.email, status)
+            if (emailError) {
+                return "Email service failed."
```

Comment:
The whole operation cannot fail if one or more emails fail. Here's why. Let's say that I have 3 hackers to send emails to: A, B, C. A and B are successful, but C is unsuccessful (server failure or something). In this case, we would return "Failure! We messed up." which essentially tells the client to retry the whole request, i.e. try sending the emails to A,B, and C. This is bad, because we've successfully sent emails to A and B. We would be spamming them! 

I recommend keeping a dictionary of successes / failures (or just of failures):
`{ A: true, B: true, C: false}` or `{C: false}`.

Reply:
Also, consider whether this error message gives you enough information. if query.exec() fails and no hackers are returned, what message are you sending back? If hackers were found but emails for all of them failed, what message are you sending back? If a specific hacker's email failed to send, what are you sending back? 

Keep in mind that these error messages are what you'll be using to debug if something breaks!